### PR TITLE
#48 Bitmap decoder

### PR DESCRIPTION
**Pull request by gscdev**

### CODE REVIEW

**Comment by gscdev** [comment]
I've added a bitmap decoder based on [Load a Scaled Down Version into Memory](https://developer.android.com/topic/performance/graphics/load-bitmap#load-bitmap)

File: app/src/main/java/ai/elimu/vitabu/ui/StoryBooksActivity.java [hunk 1]
```diff
@@ -78,14 +83,9 @@ public void run() {
...
-                                    "/Android/data/" +
-                                    BuildConfig.CONTENT_PROVIDER_APPLICATION_ID +
-                                    "/files/" + Environment.DIRECTORY_PICTURES + "/" +
-                                    coverImage.getId() + "_r" + coverImage.getRevisionNumber() + "." + coverImage.getImageFormat().toString().toLowerCase());
-                            Uri imageFileUri = Uri.fromFile(imageFile);
-                            Log.i(getClass().getName(), "imageFileUri: " + imageFileUri);
-                            coverImageView.setImageURI(imageFileUri);
+                            String imageFilePath = PICTURES_PATH + coverImage.getId() + "_r" + coverImage.getRevisionNumber() + "." + coverImage.getImageFormat().toString().toLowerCase();
+                            Log.i(getClass().getName(), "imageFile: " + imageFilePath);
+                            coverImageView.setImageBitmap(decodeSampledBitmap(imageFilePath, 100, 100));
```
Comment (Keerthi4308) [inline app/src/main/java/ai/elimu/vitabu/ui/StoryBooksActivity.java:88]: This really cool to know that we can load a scaled-down version of the image into memory. Can we use this even to load images associated with each chapter once the storybook is opened? Just wondering if we can test 100 * 100 px would be good enough for any device using the mobile app version (say tablet) or we can go even lower?

**Review by jo-elimuai — APPROVED** [review]
Here on an 8" tablet and 7" tablet for comparison:

device-2020-11-29-201838

device-2020-11-29-202547

It seems like the bitmap compression has problems with certain types of images. But the problem with the lagging has been solved, which was the main purpose of this pull request 👍

File: app/src/main/java/ai/elimu/vitabu/ui/StoryBooksActivity.java (hunk 1 above)
Comment (nya-elimuai) [reply app/src/main/java/ai/elimu/vitabu/ui/StoryBooksActivity.java:88]: @Keerthi4308 Actually, when images are extracted from a storybook ePUB (or uploaded manually), they are scaled down to 640px before being stored in the webapp's database: https://github.com/elimu-ai/webapp/blob/master/src/main/java/ai/elimu/util/ImageHelper.java

Once the storybook is opened, these original (not compressed) 640px images are displayed to the child in the storybook (Vitabu) app and in other Android apps.

device-2020-11-29-204051

Our goal is to come up with a solution that works well on all display sizes ranging from 6" (smartphones) to 10" (tablets). Perhaps storing multiple variants (resolutions) of the same image in the webapp's database would be useful?

**Review by nya-elimuai — APPROVED** [review]

**Pull request merged**